students/BrandonHenson/Lesson4/mailroom_2.py

```diff
@@ -1,125 +1,84 @@
+        fileobj = open(filename, 'w')
+        total = sum(values)
+        numgifts = len(values)
+        fileobj.write("Dear {}\nThank you for your {} generous donations \
+totaling ${}\nThe money will be put to good use.\n\n\
+        Sincerely, \n                -The Team".format(key, numgifts, total))
+        fileobj.close()
+
+    main_menu()
 
 
 def main_menu():
-    menu_answer = input(" Enter '1' for Send a thank you\n Enter '2' for \
-Create a report\n Enter '3' for\
- quit\n")
-    if menu_answer == '1':
-        menu_1()
-    elif menu_answer == '2':
-        menu_2()
-    elif menu_answer == '3':
-        exit()
-    else:
-        print("Enter 1, 2, or 3")
-        main_menu()
+    arg_dict = {
+        '1': menu_1,
+        '2': menu_2,
+        '3': email_all,
+        '4': exit
+    }
+
+    prompt = '\nSelect an option:\n' \
+             '[1] Send a Thank You\n' \
+             '[2] Create a Report\n' \
+             '[3] Send letters to everyone\n' \
+             '[4] Exit\n' \
+
+    menu_selection(prompt, arg_dict)
+
+
+def menu_selection(prompt, arg_dict):
+    try:
+        while True:
+            response = input(prompt)
+            arg_dict[response]() == ''
+    except KeyError as e:
+        print("Enter 1,2,3, or 4")
+# If the user (you) selects Send a Thank You, prompt for a Full Name
+
 
-# If the user (you) selects ‘Send a Thank You’, prompt for a Full Name
 def menu_1():
-    fullname = input("Enter a full name\n")
-# If the user types ‘list’, show them a list of the donor names and re-prompt
+    fullname = input("Enter a full name or 'list' to view all\n")
     if fullname.lower() == 'list':
         print(name_list)
         menu_1()
-    elif fullname in name_list:
-        nameindex = (name_list.index(fullname))
+    elif str(fullname) in donor_history:
         amount = float(input("Donation amount? \n"))
-        donor_history[nameindex].append(amount)
-        print("Thank you {} for your donation of ${}".format(fullname, amount))
-    elif fullname not in name_list:
-        new_profile = []
-        new_profile.append(fullname)
-        name_list.append(fullname)
+        newvalue = donor_history.get(fullname) + [amount]
+        donor_history[fullname] = newvalue
+        print("Dear {}\nThank you for your generous donation of ${}\nIt will \
+ be put to good use.\n\n Sincerely,\n\
+        -The Team".format(fullname, amount))
+    elif str(fullname) not in donor_history:
         amount = float(input("Donation amount? \n"))
-        new_profile.append(amount)
-        donor_history.append(new_profile)
-        nameindex = (name_list.index(fullname))
-        print("Thank you {} for your donation of ${}".format(fullname, amount))
-    print()
-    print()
+        donor_history[fullname] = amount
+        print("Dear {}\nThank you for your generous donation\
+ of ${}\nIt will be put to good use.\n\n Sincerely,\n\
+        -The Team".format(fullname, amount))
     main_menu()
```

Comment:
do you really need to call `main_menu` in each function? your while True loop should take care of that